shared/src/metabase/mbql/schema.cljc

```diff
@@ -321,7 +321,7 @@
 
 (def BinningStrategyName
   "Schema for a valid value for the `strategy-name` param of a [[field]] clause with `:binning` information."
-  (s/enum :num-bins :bin-width :default))
+  (s/enum :num-bins :bin-width :default :dont-bin))
```

Comment:
This isn't valid or understood in legacy MBQL so I'm not sure if adding this makes a ton of sense TBH. The canonical way to not bin something is to just not include `:binning` in the Field options... I'm worried this will potentially cause problems down the road if something doesn't recognize 

```clj
[:field {...} 1]
```

and

```clj
[:field {:binning {:strategy :dont-bin}, ...} 1]
```

Reply:
Can't we just include a version of the Field without binning options to represent the "don't bin" option?

Reply:
Oh sorry I missed that part of the PR description where you said you tried this. Let me re-read it

Reply:
Is there a meaningful difference between when someone actively chooses `Don't bin` vs. when no they haven't chosen anything? If there is, does there need to be? Seems like maybe we could simplify the frontend a bit and not need to distinguish between the two?

I'm concerned that because `:dont-bin` doesn't exist in legacy MBQL/the QP to convert back to legacy you'll just have to strip it out to make things work which means if you reload the question then it's just going to go back to looking like you haven't picked anything yet. But if that's currently set in the frontend then I guess that would have been the case anyway

Reply:
I'm a little worried about setting a precedent of encoding query builder visual state in the query itself, because as far as something like the QP goes `:dont-bin` is meaningless and not really part of the query we're compiling

Reply:
> Is there an important difference in the UI between when someone actively chooses Don't bin vs. when no they haven't chosen anything? If there is, does there need to be?

Good question. This PR exists, because not distinguishing the two was considered a bug (#31399).

> I'm concerned that because :dont-bin doesn't exist in legacy MBQL/the QP to convert back to legacy you'll just have to strip it out to make things work

I adjusted the binning middleware to treat it as no binning. I was hoping this was enough to ensure compatibility. Is that wrong?

Reply:
> I was hoping this was enough to ensure compatibility.

Yeah, that should be fine, I'm still worried about potential implications this has to reference resolution and query equality stuff. I have a gut feeling this will eventually cause a bug down the road where something gets tripped up because it doesn't consider those two versions of the `:field` clause to be the same thing